src/core/immutableMap.ts

```diff
@@ -61,14 +64,20 @@ export const mDel: MDel = <K, V>(m: ImmutableMap<K, V>, k: K) => {
   return [map]
 }
 
-export const mKeys: MKeys = <K, V>(m: ImmutableMap<K, V>) => {
-  const keys = new Set<K>()
-  m.forEach((map) => {
-    for (const key of map.keys()) {
-      keys.add(key)
-    }
+export const mForEach: MForEach = <K extends object, V>(
+  map: ImmutableMap<K, V>,
+  cb: (value: V, key: K, map: Map<K, V>) => void
+) => {
+  const seen = new WeakSet()
```

Comment:
```suggestion
  const seen = new WeakSet<K>()
```
Just to make it so with a little confidence.